### DISCUSSION TITLE
Strange error: `You don't have any datasource defined in your schema.prisma.`

### DISCUSSION THREAD

**Discussion by reza-ebrahimi** [prisma/prisma#20130] [Q&A] [answered]
### Bug description

I'm getting the following error after install `dotenv-cli` package. I'm really not sure that `dotenv-cli` package caused the error, but before than installing that everything was working fine.

Nothing is changed in my schema file, and there is a datasource as before.

Error: 
```bash
$ npx prisma generate
Environment variables loaded from .env
Prisma schema loaded from schema.prisma
Error: 
You don't have any datasource defined in your schema.prisma.
You can define a datasource like this:

datasource db {
  provider = "postgresql"
  url      = env("DB_URL")
}

More information in our documentation:
https://pris.ly/d/prisma-schema
```

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: Ubuntu 
- Database: Postgres 
- Node.js version: v20.3.0 

### Prisma Version

```
prisma                  : 4.16.2
@prisma/client          : 4.16.2
Current platform        : debian-openssl-1.1.x
Query Engine (Node-API) : libquery-engine 4bc8b6e1b66cb932731fb1bdbbc550d1e010de81 (at ../../node_modules/@prisma/engines/libquery_engine-debian-openssl-1.1.x.so.node)
Migration Engine        : migration-engine-cli 4bc8b6e1b66cb932731fb1bdbbc550d1e010de81 (at ../../node_modules/@prisma/engines/migration-engine-debian-openssl-1.1.x)
Format Wasm             : @prisma/prisma-fmt-wasm 4.16.1-1.4bc8b6e1b66cb932731fb1bdbbc550d1e010de81
Default Engines Hash    : 4bc8b6e1b66cb932731fb1bdbbc550d1e010de81
Studio                  : 0.484.0
```

**Comment by janpio** [comment]
In a default, vanilla project with Prisma the output of the CLI is actually expected to say `✔ Your Prisma schema was created at prisma/schema.prisma`.

Where are you running `npx prisma generate`? Where is your `schema.prisma` file located related to that? Is there maybe another `schema.prisma` file (which does not have a datasource defined) local to where you run the command?

**Reply by nurul3101** [reply]
@reza-ebrahimi Did you had a chance to look at Jan's response? Please let us know in case you still need assistance. If not, please consider marking the discussion as answered/closed.

**Reply by nurul3101** [reply]
We are closing this Discussion due to inactivity. Please open a new Discussion if you have a similar problem 🙏

**Reply by reza-ebrahimi** [reply, marked as answer]
Sorry for late reply, I couldn't find the root cause, but it fixed with a combination of reverting the latest changes + deleting `node_modules` directory + deleting `.next` directory and rerunning project.
